sessions/react-state-3/journal-app-favorite-button-shared-state/README.md

```diff
@@ -0,0 +1,58 @@
+# React State: Journal App Favorite Button with shared state
+
+Currently our the favorite feature of your journal app is using a local state. We want to store this information in the shared state alongside the other entry data.
+
+> 💡 You can use this template as a starting point. But if you are far enough along with your own Journal App, please use that instead.
+
+## Task
+
+First we need to make sure all entries in state contain information about the favorite button. Locate the component holding the shared state for all entries. Here it is the `App` component.
+
+- Add the key `isFavorite` with the value `false` to each entry in the `initialEntries` array.
+- Add the key `isFavorite` with the value `false` to a newly created entry in the function `handleAddEntry`.
+
+Now let's wire up this information stored in state with the favorite button.
+
+Locate the component with the favorite button. Here it's called `FavoriteButton`.
+
+- Accept a new prop in this component called `onToggleFavorite`. This event should be triggered when users press the favorite button.
+- Accept a new prop called `isFavorite`.
+- Accept a new prop called `id`.
+- Remove the local state `isFavorite` from this component.
+- Change the button's `onClick` prop to call `onToggleFavorite` and pass `id` with this call as first argument.
+
+Locate the component that renders a single entry including the favorite button. Here it's `Entry`.
+
+- In this component accept the three props too: `onToggleFavorite`, `isFavorite`, `id`.
+- Pass all three props to the `FavoriteButton` component.
+
+Locate the component that renders the list of all entries. Here it's `EntrySection`
+
+- Accept a new prop `onToggleFavorite`
+- Adjust the integration of the `Entry` component to pass the `onToggleFavorite` prop
+- In addition pass each entry's `id` and `isFavorite` information to the `Entry` component as prop
+
+Go back to the `App` component.
+
+- Create a new function called `handleToggleFavorite` which accepts `id` as parameter
+- Inside this function search for the entry matching the `id` and flip the value of the `isFavorite` key. (hint: you can use the array method `map` for this)
+- Update the state by calling `setEntries`.
```

Comment:
This was the only confusing part to me: I thought you want me to do this in separate steps:
1. use `map` first for searching for the correct entry and update its `isFavorite` key and then
2. update state (but the handout tells me to use `map` as well for replacing an object.

I am not sure how we would do these steps separately if we want to use `map` for both, but I guess I just don't see an easier way right now.

Just to be sure: is the following code snippet the way we want it? It works, but for me, this is rather one step than two. Do I miss something or can we fix this with rephrasing the instruction?

```js
setEntries(
      entries.map((entry) => {
        if (entry.id === id) return { ...entry, isFavorite: !entry.isFavorite };
        return entry;
      })
    );
```

Reply:
You are right! The task description was confusing. I changed like this


> - Create a new function called `handleToggleFavorite` which accepts `id` as parameter.
> - Inside this function update the state by calling `setEntries`. To create a new state value iterate over the `entries` array and search for the entry matching the `id`. Flip the value of it's `isFavorite` key. (hint: you can use the array method `map` for this)

I hope the goal is more clear now. Your code snippets reflects exactly what the students do!